src/xray/jest-xray-environment.ts

```diff
@@ -0,0 +1,37 @@
+import NodeEnvironment from "jest-environment-node";
+
+class JestXRayEnvironment extends NodeEnvironment {
+
+  override async setup() {
+    await super.setup();
+  }
+
+  async handleTestEvent(event: any) {
+    if (event.name === "test_start") {
+      let testNames = [];
+      let currentTest = event.test;
+
+      while (currentTest) {
```

Comment:
Or you stop if you have the first two entries of `testNames`. Shouldn't we add this to the condition?